_msprimemodule.c

```diff
@@ -2109,9 +2095,19 @@ RecombinationMap_get_positions(RecombinationMap *self)
         handle_library_error(err);
         goto out;
     }
-    ret = convert_float_list(positions, size);
+    ret = PyArray_SimpleNewFromData(1, &size, NPY_FLOAT64, positions);
```

Comment:
I think it would be simpler to do something like
```C
position_array =  PyArray_SimpleNewFromData(1, &size, NPY_FLOAT64);
if (position_array == NULL) { etc}
err = recomb_map_get_positions(self->recomb_map, PyArray_DATA(position_array));
```
Basically, we let numpy allocate the memory to store the array.

The capsule stuff is very clever, but this is simpler/more robust.